include/boost/geometry/formulas/area_formulas.hpp

```diff
@@ -442,58 +447,52 @@ class area_formulas
 
         // Basic trigonometric computations
 
-        CT tan_bet1 = tan(get_as_radian<1>(p1)) * one_minus_f;
-        CT tan_bet2 = tan(get_as_radian<1>(p2)) * one_minus_f;
-        CT cos_bet1 = cos(atan(tan_bet1));
-        CT cos_bet2 = cos(atan(tan_bet2));
-        CT sin_bet1 = tan_bet1 * cos_bet1;
-        CT sin_bet2 = tan_bet2 * cos_bet2;
-        CT sin_alp1 = sin(alp1);
-        CT cos_alp1 = cos(alp1);
-        CT cos_alp2 = cos(alp2);
-        CT sin_alp0 = sin_alp1 * cos_bet1;
+        CT const tan_bet1 = tan(lat1r) * one_minus_f;
+        CT const tan_bet2 = tan(lat2r) * one_minus_f;
+        CT const cos_bet1 = cos(atan(tan_bet1));
+        CT const cos_bet2 = cos(atan(tan_bet2));
+        CT const sin_bet1 = tan_bet1 * cos_bet1;
+        CT const sin_bet2 = tan_bet2 * cos_bet2;
+        CT const sin_alp1 = sin(alp1);
+        CT const cos_alp1 = cos(alp1);
+        CT const cos_alp2 = cos(alp2);
+        CT const sin_alp0 = sin_alp1 * cos_bet1;
 
         // Spherical term computation
 
-        CT sin_omg1 = sin_alp0 * sin_bet1;
-        CT cos_omg1 = cos_alp1 * cos_bet1;
-        CT sin_omg2 = sin_alp0 * sin_bet2;
-        CT cos_omg2 = cos_alp2 * cos_bet2;
-        CT cos_omg12 =  cos_omg1 * cos_omg2 + sin_omg1 * sin_omg2;
+        CT const sin_omg1 = sin_alp0 * sin_bet1;
+        CT const cos_omg1 = cos_alp1 * cos_bet1;
+        CT const sin_omg2 = sin_alp0 * sin_bet2;
+        CT const cos_omg2 = cos_alp2 * cos_bet2;
+        CT const cos_omg12 =  cos_omg1 * cos_omg2 + sin_omg1 * sin_omg2;
         CT excess;
 
         bool meridian = get<0>(p2) - get<0>(p1) == CT(0)
               || get<1>(p1) == CT(90) || get<1>(p1) == -CT(90)
               || get<1>(p2) == CT(90) || get<1>(p2) == -CT(90);
 
-        if (!meridian && cos_omg12 > -CT(0.7)
-                      && sin_bet2 - sin_bet1 < CT(1.75)) // short segment
+        if (!meridian && cos_omg12 > -CT(0.7071)
```

Comment:
Is this supposed to be 1/sqrt(2)? Perhaps use a constant for this?

Reply:
You are right, this is to select between trapezoid formula and azimuth subtraction (i.e. alp2 - alp1) for spherical excess computation. 

With trapezoid formula we have the following relative errors (using the dataset from https://zenodo.org/record/32156#.WOyaMrUmv7C

maximum legnth | 2 | 10 | 100 | 1000 | 2000 | 1000000 | 2000000 | 5000000 | 7500000 | 10000000 | 12500000 | 15000000 | 17500000 | 19500000 | 20000000
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
andoyer | 0.002051124711 | 5.00E-06 | 1.80E-07 | 6.16E-06 | 1.30E-09 | 1.37E-05 | 5.49E-05 | 0.0003688702402 | 0.0009236612365 | 0.001903669355 | 0.003755516045 | 0.007758503031 | 0.01998210139 | 0.1029663252 | 0.4328622397
thomas | 0.004475880816 | 2.88E-10 | 3.86E-09 | 3.85E-10 | 1.61E-10 | 1.37E-05 | 5.49E-05 | 0.0003688700554 | 0.0009236610226 | 0.00190366909 | 0.003755515698 | 0.007758502504 | 0.0199821028 | 0.102966375 | 0.4329122004
vincenty | 0.004729452869 | 4.34E-06 | 1.63E-07 | 1.53E-07 | 1.52E-07 | 1.36E-05 | 5.49E-05 | 0.0003689026972 | 0.0009236909016 | 0.001903649421 | 0.003755454646 | 0.007757974129 | 0.0199798291 | 0.1029193871 | 0.4322189917

While with azimuth subtraction (i.e. alp2 - alp1)

maximum legnth | 2 | 10 | 100 | 1000 | 2000 | 1000000 | 2000000 | 5000000 | 7500000 | 10000000 | 12500000 | 15000000 | 17500000 | 19500000 | 20000000
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
andoyer | 0.9997504932 | 0.001255017513 | 7.88E-05 | 0.00214284374 | 9.84E-07 | 2.65E-10 | 5.72E-09 | 1.49E-09 | thomas | 1.09E-08 | 5.73E-09 | 4.34E-11 | 1.50E-10 | 3.42E-11 | 1.60E-10 | 0.06630115469
vincenty| 1 | 1.69E-07 | 6.84E-06 | 8.66E-06 | 5.48E-08 | 3.78E-08 | 3.83E-08 | 4.13E-08 | 4.77E-08 | 5.89E-08 | 9.15E-08 | 3.70E-07 | 3.22E-06 | 0.0002948330798 | 0.214268049
1.689545791 | 0.001697061092 | 9.38E-05 | 8.60E-06 | 1.59E-07 | 1.67E-07 | 3.23E-07 | 2.04E-06 | 5.61E-06 | 1.37E-05 | 3.45E-05 | 0.0001037443284 | 0.0005077352782 | 0.01102688439 | 0.1776476855

So we want to keep the best of both worlds which is not very simple by using the trigonometric thresholds and since we have already computed the azimuths in a previous step of the computation, we can compute the distance too and use it as a threshold. Thus, we have to select the former formula for segments with length < 2000 and the latter for the rest. In lack of data between 2000 and 1000000 segment lengths I am choosing 10000 as a threshold.